### DISCUSSION TITLE
`terraform destroy` does not remove LB and network

### DISCUSSION THREAD

**Discussion by loxy** [kube-hetzner/terraform-hcloud-kube-hetzner#263] [Q&A] [answered]
Hi!

Really cool Terraform module. Thank you all!

I tried to remove everything from Hetzner again with `terraform destroy`. So far it looks good, but the load balancer and the network get not removed.

```sh
module.kube-hetzner.hcloud_network_subnet.agent[1]: Still destroying... [id=1894766-10.1.0.0/16, 10s elapsed]
...
module.kube-hetzner.hcloud_network_subnet.agent[1]: Still destroying... [id=1894766-10.1.0.0/16, 9m30s elapsed]
...
```

This goes on an on... 

Do I miss something? Or is it not possible to remove everything via `terraform destroy`?

Best regards,
Kersten

**Comment by mysticaltech** [comment, marked as answer]
@loxy just delete the LB that is created on the request of Traefik, and re-run the destroy. Either via the UI, or with hcloud, but running `hcloud load-balancer delete <cluster-name>-traefik`, you can also use `hcloud load-balancer list` to find its name.

_This is explained in the readme in the destroy section._

**Reply by mysticaltech** [reply]
This is because it's not Terraform that creates the LB.

**Reply by loxy** [reply]
Ok. Thank you. That's what I have done. So, it is somehow Traefik who creates the LB and Terraform is not aware of that?
And I assume that this is the reason why the network is not deleted, because the LB is still alive?
Is it possible to re-use the LB (when it was not destroyed)?

**Reply by mysticaltech** [reply]
Yes to your first questions. The last one, no, onces destroyed, it gone forever.

Traefik ask to the CCM, and the CCM provides the LB.

Because you have to destroy it, to destroy the cluster!

**Reply by loxy** [reply]
Understood. Thank you again!
